src/Components/Explorer/ExplorerNode.story.lua

```diff
@@ -0,0 +1,50 @@
+local constants = require(script.Parent.Parent.Parent.constants)
+local ExplorerNode = require(script.Parent.ExplorerNode)
+local fromHex = require(script.Parent.Parent.Parent.Modules.fromHex)
+local hook = require(script.Parent.Parent.Parent.hook)
+local Roact = require(script.Parent.Parent.Parent.Packages.Roact)
+local types = require(script.Parent.types)
+
+local e = Roact.createElement
+
+local childNode = {
+	icon = "story",
+	name = "TestComponent.story",
+}
+
+local parentNode = {
+	children = { childNode },
+	icon = "folder",
+	name = "TestFolder",
+}
+
+local function ExplorerNodeStory(_, hooks: any)
+	local state, set = hooks.useState(nil)
+	local onNodeActivated = hooks.useCallback(function(node: types.Node)
+		if node.name:match(constants.STORY_NAME_PATTERN) then
+			if node == state then
+				set(nil)
+			else
+				set(node)
+			end
```

Comment:
Looks good! Make sure you don't reference state directly when calculating new state based on previous state. `useCallback` is also likely doing more harm than good here, especially with luau closure optimizations.
```suggestion
			set(function(prevState)
			    return if node == prevState then nil else node
			end)
```